src/main/java/com/avispl/symphony/dal/avdevices/audionetworkinterface/qsc/undnemo/QSCUndnemoCommunicator.java

```diff
@@ -0,0 +1,805 @@
+				stop();
+			}
+		}
+
+		/**
+		 * Triggers main loop to stop
+		 */
+		public void stop() {
+			inProgress = false;
+		}
+	}
+
+	/**
+	 * Executor that runs all the async operations, that {@link #channelDataLoader} is posting
+	 */
+	private static ExecutorService executorService;
+
+	/**
+	 * Runner service responsible for collecting data
+	 */
+	private QSCChannelDataLoader channelDataLoader;
+
+	/**
+	 * List of channel info
+	 */
+	private List<ChannelInfo> channelInfoList = Collections.synchronizedList(new ArrayList<>());
+
+	/**
+	 * List error message occur while fetching channel infos
+	 */
+	private Set<String> channelErrorMessagesList = Collections.synchronizedSet(new LinkedHashSet<>());
+
+	/**
+	 * This field is used to prevent fetching unnecessary data when perform {@link QSCUndnemoCommunicator#controlProperty(ControllableProperty)}
+	 */
+	private boolean isActiveChannelControl = false;
+
+	/**
+	 * This field is used to check if the {@link QSCUndnemoCommunicator#getMultipleStatistics()} -
+	 * is called after {@link QSCUndnemoCommunicator#controlProperty(ControllableProperty)}
+	 */
+	private boolean isGetMultipleStatsAfterControl = false;
+
+	/**
+	 * Local extended statistics
+	 */
+	private ExtendedStatistics localExtendedStatistics;
+
+	/**
+	 * Adapter Properties - (Optional) filter option: string of channel indexes (separated by commas)
+	 */
+	private String channelIndex;
+
+	/**
+	 * Retrieves {@code {@link #channelIndex}}
+	 *
+	 * @return value of {@link #channelIndex}
+	 */
+	public String getChannelIndex() {
+		return channelIndex;
+	}
+
+	/**
+	 * Sets {@code channelIndex}
+	 *
+	 * @param channelIndex the {@code java.lang.String} field
+	 */
+	public void setChannelIndex(String channelIndex) {
+		this.channelIndex = channelIndex;
+	}
+
+	/**
+	 * Constructor set command error and success list that is required by {@link UDPCommunicator}
+	 */
+	public QSCUndnemoCommunicator() {
+		// set buffer length because the response may exceed the default value.
+		this.setBufferLength(100);
+
+		this.setCommandSuccessList(Collections.singletonList(UDPCommunicator.getHexByteString(new byte[] { (byte) 0x00, 0x00, (byte) 0x00 })));
+		this.setCommandErrorList(Collections.singletonList(
+				UDPCommunicator.getHexByteString(new byte[] { (byte) 0x00, 0x00, (byte) 0x00 })
+		));
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	protected void internalInit() throws Exception {
+		if (logger.isDebugEnabled()) {
+			logger.debug("Internal init is called.");
+		}
+		super.internalInit();
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	protected void internalDestroy() {
+		if (logger.isDebugEnabled()) {
+			logger.debug("Internal destroy is called.");
+		}
+
+		if (channelDataLoader != null) {
+			channelDataLoader.stop();
+			channelDataLoader = null;
+		}
+
+		if (executorService != null) {
+			executorService.shutdownNow();
+			executorService = null;
```

Comment:
Is it necessary to re-assign it to null?

Reply:
Updated.